hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/ServiceFactory.java

```diff
@@ -315,10 +310,65 @@ private ServiceFactory(
      */
     @Nonnull
     public ServiceFactory<C, S> withUnorderedAsyncResponses() {
-        return new ServiceFactory<>(
-                createContextFn, createServiceFn, destroyServiceFn, destroyContextFn,
-                isCooperative, maxPendingCallsPerProcessor, false
-        );
+        ServiceFactory<C, S> copy = clone();
+        copy.orderedAsyncResponses = false;
+        return copy;
+
+    }
+
+    /**
+     * Attaches a file to this service factory. It will become a part of the
+     * Jet job and available to {@link #createContextFn()} as {@link
+     * ProcessorSupplier.Context#attachedFile
+     * procSupplierContext.attachedFile(file.toString())}.
+     *
+     * @return a copy of this factory with the file attached
+     *
+     * @since 4.0
+     */
+    @Nonnull
+    public ServiceFactory<C, S> withAttachedFile(@Nonnull File file) {
+        if (!file.isFile()) {
```

Comment:
we don't support ID here which is contradictory with how it works when doing it through JobConfig. Wouldn't it make sense to always require ID here, and then pythonService can add its own special id for these files?